pkgs/applications/version-management/github-desktop/default.nix

```diff
@@ -0,0 +1,100 @@
+{ darkMode ? false, stdenv, fetchurl, dpkg, makeWrapper
+, alsaLib, atk, cairo, cups, curl, dbus, expat, fontconfig, freetype, glib
+, gnome2, gtk3, gdk_pixbuf, libnotify, libxcb, nspr, nss, pango
+, systemd, xorg }:
+
+let
+
+  version = "1.5.1";
+
+  rpath = stdenv.lib.makeLibraryPath [
+    alsaLib
+    atk
+    cairo
+    cups
+    curl
+    dbus
+    expat
+    fontconfig
+    freetype
+    glib
+    gnome2.GConf
+    gdk_pixbuf
+    gtk3
+    pango
+    libnotify
+    libxcb
+    nspr
+    nss
+    stdenv.cc.cc
+    systemd
+
+    xorg.libxkbfile
+    xorg.libX11
+    xorg.libXcomposite
+    xorg.libXcursor
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXi
+    xorg.libXrandr
+    xorg.libXrender
+    xorg.libXtst
+    xorg.libXScrnSaver
+  ] + ":${stdenv.cc.cc.lib}/lib64";
+
+  src =
+    if stdenv.hostPlatform.system == "x86_64-linux" then
+      fetchurl {
+        url = "https://github.com/shiftkey/desktop/releases/download/release-1.5.1-linux2/GitHubDesktop-linux-${version}-linux2.deb";
+        sha256 = "0nbgv920l9l49046yv8nal990p6nxgsh6fw18bxpflg9h6qi4afq";
+      }
+    else
+      throw "Slack is not supported on ${stdenv.hostPlatform.system}";
+
+in stdenv.mkDerivation {
+  name = "github-desktop-${version}";
+
+  inherit src;
+
+  buildInputs = [
+    dpkg
+    gtk3  # needed for GSETTINGS_SCHEMAS_PATH
+  ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  unpackPhase = "true";
+  buildCommand = ''
+    mkdir -p $out
+    dpkg -x $src $out
+
+    mv $out/usr/share $out/share
+    mv $out/opt/GitHub\ Desktop $out/libexec
+    rmdir $out/usr $out/opt
+
+    chmod -R g-w $out
+
+    # patch the binaries
```

Comment:
did you try if adding `autopatchelfHook` to `nativeBuildInputs` works, so you don't need to manually `patchelf` and assemble rpath…

Reply:
Not yet, thankyou for the recommendation.

Reply:
Moving to `autopatchelfHook`made a substantial difference to the cleanliness/maintainability of the expression. It's currently failing atm in the patching phase. Have pushed code for the night and intend to pick this up again soon.